src/construction.cpp

```diff
@@ -472,8 +485,13 @@ void construction_menu()
 
         const std::string action = ctxt.handle_input();
         const long raw_input_char = ctxt.get_raw_input().get_first_input();
-
-        if( action == "DOWN" ) {
+        if( action == "FILTER" ){
+            update_info = true;
+            update_cat = true;
+            tabindex = 9;
```

Comment:
I don't know how to do it in real time, as my experience relies entirely on `string_input_popup`.

Is there a place in the code which you have in mind?